### ISSUE TITLE
[2021] Diversifying Dialog Generation via Adaptive Label Smoothing

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#144]
## 0. 論文

| タイトル | Diversifying Dialog Generation via Adaptive Label Smoothing |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2105.14556 |
| 著者    | Yida Wang, Yinhe Zheng, Yong Jiang, Minlie Huang |
| 投稿日付 | ACL2021 |

## 1. どんなもの？

![スクリーンショット 2021-08-16 14 19 37](https://user-images.githubusercontent.com/1255485/129514677-75450198-71a3-4ce8-8812-baad3b1057ea.png)

対話システムにおいて、生成の多様性が低い問題の原因の一つとして (one-hot) hard target を学習に使っていることが挙げられる。別の1層の decoder を用意してターゲットの分布を予測し、それを本来の hard target と合わせてメインの decoder の予測に使用する Adaptive Label Smoothing (AdaLabel) を提案。distillation と似た枠組みだが、teacher となる auxiliary decoder がかなり小さく、かつ encoder を共有して end-to-end に学習可能。
多くのベースラインモデルと比較し、fluency や coherency を維持したまま生成の多様性を大きく改善。

## 2. 先行研究と比べてどこがすごい？

- auxiliary decoder を用意し、その予測を soft target として使用する
- hard target と soft target の割合はダイナミックに変更し、元の予測確率値が高いときはそちらを多く利用するアイディア

## 3. 技術や手法のキモはどこ？

![スクリーンショット 2021-08-16 18 51 17](https://user-images.githubusercontent.com/1255485/129545339-72ea6cca-c7a1-449e-b3a5-d0e592c87899.png)

- モデル
    - 上図のように、通常の decoder に加えて auxiliary decoder (1層、bi-directional) を用意し、target を予測
        - 本来のターゲット (hard target) と aux decoder の推論で得た soft target を組み合わせた分布を decoder で予測する形で学習
        - 単に label smoothing で均等に確率を割り振るのではなく、bi-directional な aux decoder の予測を基に soft-target を作成することで改善
    - hard-target と soft-target は動的に割合を調整しながら組み合わせている
        - q を hard-target (label), v を soft-target (aux decoder の予測) として、decoder のターゲットとなる q' は
            - ![スクリーンショット 2021-08-16 19 12 02](https://user-images.githubusercontent.com/1255485/129548022-73812995-1ee9-402c-b694-1ff831a2eea9.png)
        - ここで ε は、decoder の予測確率の最も高い値 p_{max} を基に次で計算する
            - ![スクリーンショット 2021-08-16 19 14 13](https://user-images.githubusercontent.com/1255485/129548266-5b187ad0-6dd8-449f-95f6-34dfe0f975f5.png)
            - すなわち、予測確率で最も高いものと同じくらいの重みで hard-target を利用する (上図参照)
        - 提案手法ではこれにさらに予測から算出する α で重み付けして調整 (論文参照)
    - aux decoder は bi-directional なので、下図のように target 部分のみ mask するような attention mask を使用する
    -   ![スクリーンショット 2021-08-16 19 16 33](https://user-images.githubusercontent.com/1255485/129548559-804940f4-3399-4bab-a8e4-4657f3185873.png)

## 4. どうやって有効だと検証した？

多くのベースラインモデル(9個)と比較し、全体的に提案手法が高い性能になっている。

- ベンチマークデータセット
    - DailyDialog と OpenSubtitles
    - それぞれ、1つの発話が 100 tokens 以上のものは消してるらしい
- ベースラインモデル
    - LS: one-hot を弱めるために label smoothing を使用 (Szegedy et al., 2016)
    - FL: よく分類できているクラスの重みを下げるため focal loss を使用 (Lon et al., 2017)
    - FACE (Jiang et al., 2019): frequency-aware cross-entropy loss を使用, よく出るトークンは小さい loss になる
    - F^2 (Choi et al., 2020): トークンの頻度に応じてターゲットの分布を factorize
    - CP (Pereyret et al., 2017): 学習字に confidence penalty term を追加, over-confident な予測にペナルティ
    - UL (Welleck et al., 2020): よく出るトークンにペナルティを与えるため unlikelihood loss を追加
    - NL (He and Glass, 2020): UL の unlikelihood loss の代わりに negative loss なるものを導入
    - D2GPo (Li et al., 2019): MLE に data-dependent gaussian prior objective を追加する、ターゲットでないトークンにも loss を付与
    - CE: 普通の cross entropy で学習した Transformer
- 学習・モデル
    - Transformer ベースの6 layers, hidden size 512, 8 attention heads (encoder/decoder)
    - auxiliary decoder は上と同様で 1 layer
- 結果
    - 自動評価指標 Dist, Entropy, LF (Low Frequency Token Ratio), BLEU に加えて人手評価 Fluency, Coherency, Informativeness と overall preference で評価。人手評価は2つの返答の比較形式。
    - 自動評価指標では提案手法が diversity 系の指標で改善
    - ![スクリーンショット 2021-08-16 18 31 31](https://user-images.githubusercontent.com/1255485/129542546-fe44fa3f-3aa1-4252-b767-b8c1d5cdb063.png)
    - 人手評価でも全ベースラインの中で最も良い結果に。下記表は win rate - lose rate の値
    - ![スクリーンショット 2021-08-16 18 35 53](https://user-images.githubusercontent.com/1255485/129543205-6758839a-04a3-4788-a382-c764ef12ed33.png)

## 5. 議論はある？

- auxiliary decoder は1層でも bi-directional なので普通の decoder より良い性能。
- 予測確率の分布をグラフ化すると、提案手法以外は確率1にspikeがあるが、提案手法はスムーズな分布 (Fig. 4)
- レアなトークンの生成確率も提案手法の方が高いことを確認できた
- future work で他のテキスト生成タスクに適用
